crates/shared/src/request_sharing.rs

```diff
@@ -28,14 +30,22 @@ pub type BoxRequestSharing<Request, Response> =
 /// A boxed shared future.
 pub type BoxShared<T> = Shared<BoxFuture<'static, T>>;
 
-impl<Request, Fut: Future> Default for RequestSharing<Request, Fut> {
-    fn default() -> Self {
+impl<Request, Fut: Future> RequestSharing<Request, Fut> {
+    pub fn labelled(request_label: String) -> Self {
         Self {
             in_flight: Default::default(),
+            request_label,
         }
     }
 }
 
+/// Returns a shallow copy (without any pending requests)
+impl<Request, Fut: Future> Clone for RequestSharing<Request, Fut> {
+    fn clone(&self) -> Self {
+        Self::labelled(self.request_label.clone())
```

Comment:
I think it would be easier to understand if `Self` is created in place instead of going through `Self::labelled()`.